x-pack/plugins/ingest_manager/server/errors.ts

```diff
@@ -4,26 +4,20 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+/* eslint-disable max-classes-per-file */
 export class IngestManagerError extends Error {
-  public type: IngestManagerErrorType;
-  public message: string;
-
-  constructor(type: IngestManagerErrorType, message: string) {
+  constructor(message?: string) {
     super(message);
-    this.type = type;
-    this.message = message;
+    this.name = this.constructor.name; // for stack traces
   }
 }
 
 export const getHTTPResponseCode = (error: IngestManagerError): number => {
-  switch (error.type) {
-    case IngestManagerErrorType.RegistryError:
-      return 502; // Bad Gateway
-    default:
-      return 400; // Bad Request
+  if (error instanceof RegistryError) {
+    return 502; // Bad Gateway
+  } else {
+    return 400; // Bad Request
   }
 };
 
-export enum IngestManagerErrorType {
-  RegistryError,
-}
+export class RegistryError extends IngestManagerError {}
```

Comment:
I chose to extend `IngestManagerError` here so that it gets the benefits of the stack trace naming from https://github.com/elastic/kibana/pull/69966/files#diff-305aab707fa2746fa91c6bec9e78c0b7R11 and remain an `IngestManagerError`

However, even if it did `extend Error` we could make a type like

```typescript
type IngestManagerErrorType = IngestManagerError | RegistryError
```

This is where my mind is heading now. Files will define/export Errors where they are used (or perhaps lifted to `common` or `server`), and then those can be imported and combined into other types